src/ktopt_planning_context.cpp

```diff
@@ -177,6 +180,42 @@ void KTOptPlanningContext::solve(planning_interface::MotionPlanResponse& res)
   // TODO: These should be parameters
   trajopt.AddDurationConstraint(0.5, 5);
 
+  // process path_constraints
+  if (!req.path_constraints.position_constraints.empty())
+  {
+    for (const auto& position_constraint : req.path_constraints.position_constraints)
+    {
+      // Extract the bounding box's center (primitive pose)
+      const auto& primitive_pose = position_constraint.constraint_region.primitive_poses[0];
+      Eigen::Vector3d box_center(primitive_pose.position.x, primitive_pose.position.y, primitive_pose.position.z);
+
+      // Extract dimensions of the bounding box from
+      // constraint_region.primitives Assuming it is a box
+      // (shape_msgs::SolidPrimitive::BOX) and has dimensions in [x, y, z]
+      const auto& primitive = position_constraint.constraint_region.primitives[0];
+      if (primitive.type == shape_msgs::msg::SolidPrimitive::BOX)
```

Comment:
Would prefer the reverse: If the primitive is *not* a box, raise a warning and continue the loop; else, carry on.

This, in addition to being more readable, will save you an indent level.